src/led.h

```diff
@@ -15,4 +14,74 @@
 #define LED_h   0x12
 #define LED_dp  0x13
 
-void filldisplay(uint8_t pos, uint8_t val, __bit dp);
+const uint8_t __at (0x1000) ledtable[]
```

Comment:
I think you have quite an interesting and useful "hack" here. A few observations (some of which you had discovered):

- flash and eeprom are likely the same flash, just with some "partitioning" around them. (I'm not sure why they are selling this eeprom vs flash thing.. seems to be artificial boundary) 
- because of this you can access directly through pointers
- sdcc will happily start code segments (such as your led table, or as I have found out, even functions) wherever you ask it to (e.g., __at (0x1000) [which is start of "eeprom" block]) and generate an intel hex which has instructions to program this range.
- when this hex file is loaded into stc-isp tool, it recognizes this is bigger than the "stated" part specs, and throws and error, _however_, stcgal throws a warning but continues on and attempts to program this range, which bootloader is happy to do.

The overall, very interesting result, is that you can load code/data/whatever into "eeprom" just as flash/code and execute it just the same. 
The only concern is that the bootloader will not overwrite the last 7 bytes of "flash" section which is reserved for ID. 
So perhaps there is an interesting hack here to let sdcc use the full 5k code space automatically:
1. tell sdcc to use 5k code size,
2. somehow "reserve" the last 7 bytes up to 0x1000,
3. of course, use stcgal for programming (stc-isp wont do it).